player/command.c

```diff
@@ -5528,6 +5528,24 @@ static void cmd_dump_cache_ab(void *p)
                  cmd->args[0].v.s);
 }
 
+static void cmd_vo_maximize(void *p)
+{
+    struct mp_cmd_ctx *cmd = p;
+    struct MPContext *mpctx = cmd->mpctx;
+
+    if (mpctx->video_out)
+        vo_control(mpctx->video_out, VOCTRL_MAXIMIZE, 0);
+}
+
+static void cmd_vo_minimize(void *p)
+{
+    struct mp_cmd_ctx *cmd = p;
+    struct MPContext *mpctx = cmd->mpctx;
+
+    if (mpctx->video_out)
+        vo_control(mpctx->video_out, VOCTRL_MINIMIZE, 0);
+}
+
```

Comment:
These should be properties, not command. There's already window-minimized property too.

Reply:
For maximize, I cheaped out because I didn't want to deal with the out-of-band changes to the state (which your other proposed changes addresses). For minimize, wayland makes this a write-only property. You can't query the state and don't get notified when it changes. 

[As documented here](https://www.enlightenment.org/_legacy_embed/group__iface__xdg__toplevel.html#ga813e6d50fcc67be73167510869a7e714)

Reply:
God, why is wayland so bad? I don't want that a window can minimize itself (what the fuck is the point of that? at best some software could use it to be more annoying), while on the other hand, programs _do_ benefit from knowing whether they're minimalized. The existing window-minimalized property in mpv was read-only for starters and was added because it was _useful_ for scripts to know the state.

I sure wish someone would save X11 and absolve us from the horror that is wayland.

/rant